Source/UserManagement/Domain/DataCollectors/CommandHandlers/DataCollectorCommandHandler.cs

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) 2017 International Federation of Red Cross. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+using doLittle.Runtime.Commands;
+using doLittle.Domain;
+using System;
+using Domain.DataCollectors.AggregateRoots;
+using Domain.DataCollectors.Commands;
+
+namespace Domain.DataCollectors.CommandHandlers
+{
+    public class DataCollectorCommandHandler : IDataCollectorCommandHandler
+    {
+        private readonly IAggregateRootRepositoryFor<DataCollectorManagement> _repository;
+
+        public DataCollectorCommandHandler (
+            IAggregateRootRepositoryFor<DataCollectorManagement> repository
+            )
+        {
+            _repository = repository;
+        }
+
+        public void Handle(AddDataCollector command)
+        {
+            var root = _repository.Get(command.DataCollectorId);
+            root.AddDataCollector(command);
+        }
+
+        public void Handle(AddPhoneNumberToDataCollector command)
```

Comment:
Again, no commands passed to the aggregate root.

root.AddPhoneNumber(command.PhoneNumber)

I am also a bit suspicious of the Add* and Remove* names as it feels like CRUD.  I don't know the domain well enough to say if this is correct or not though.